### PR TITLE
Add Call Filter That Prevents Nested `batch_all`

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR adds a call filter to `batch_all` which prevents users from nesting `batch_all` calls into one another.

### CODE REVIEW

**Review by gavofyork — APPROVED** [review]
would be nice to test that `batch_all(batch(batch_all(` still fails. looks good otherwise.

File: frame/utility/src/lib.rs [hunk 1]
```diff
@@ -266,7 +268,13 @@ pub mod pallet {
 				let result = if is_root {
 					call.dispatch_bypass_filter(origin.clone())
 				} else {
-					call.dispatch(origin.clone())
+					let mut filtered_origin = origin.clone();
```
Comment (athei) [inline frame/utility/src/lib.rs:271]: There is no need to clone, right? You can just move the value as it is not used later.

**Review by athei — APPROVED** [review]

File: frame/utility/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply frame/utility/src/lib.rs:271]: There is a loop

**Comment by shawntabrizi** [comment]
bot merge

**Review by apopiak — APPROVED** [review]
LGTM

**Pull request merged**